dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DataQueryRequest.java

```diff
@@ -0,0 +1,510 @@
+        return hideEmptyRows;
+    }
+
+    public boolean isHideEmptyColumns()
+    {
+        return hideEmptyColumns;
+    }
+
+    public boolean isShowHierarchy()
+    {
+        return showHierarchy;
+    }
+
+    public boolean isIncludeNumDen()
+    {
+        return includeNumDen;
+    }
+
+    public boolean isIncludeMetadataDetails()
+    {
+        return includeMetadataDetails;
+    }
+
+    public DisplayProperty getDisplayProperty()
+    {
+        return displayProperty;
+    }
+
+    public IdScheme getOutputIdScheme()
+    {
+        return outputIdScheme;
+    }
+
+    public IdScheme getInputIdScheme()
+    {
+        return inputIdScheme;
+    }
+
+    public String getApprovalLevel()
+    {
+        return approvalLevel;
+    }
+
+    public Date getRelativePeriodDate()
+    {
+        return relativePeriodDate;
+    }
+
+    public String getUserOrgUnit()
+    {
+        return userOrgUnit;
+    }
+
+    public DhisApiVersion getApiVersion()
+    {
+        return apiVersion;
+    }
+
+    public boolean isDuplicatesOnly()
+    {
+        return duplicatesOnly;
+    }
+
+    public boolean isAllowAllPeriods()
+    {
+        return allowAllPeriods;
+    }
+
+    private DataQueryRequest( DataQueryRequestBuilder builder )
+    {
+        this.dimension = builder.dimension;
+        this.filter = builder.filter;
+        this.aggregationType = builder.aggregationType;
+        this.measureCriteria = builder.measureCriteria;
+        this.preAggregationMeasureCriteria = builder.preAggregationMeasureCriteria;
+        this.startDate = builder.startDate;
+        this.endDate = builder.endDate;
+        this.order = builder.order;
+        this.skipMeta = builder.skipMeta;
+        this.skipData = builder.skipData;
+        this.skipRounding = builder.skipRounding;
+        this.completedOnly = builder.completedOnly;
+        this.hierarchyMeta = builder.hierarchyMeta;
+        this.ignoreLimit = builder.ignoreLimit;
+        this.hideEmptyRows = builder.hideEmptyRows;
+        this.hideEmptyColumns = builder.hideEmptyColumns;
+        this.showHierarchy = builder.showHierarchy;
+        this.includeNumDen = builder.includeNumDen;
+        this.includeMetadataDetails = builder.includeMetadataDetails;
+        this.displayProperty = builder.displayProperty;
+        this.outputIdScheme = builder.outputIdScheme;
+        this.inputIdScheme = builder.inputIdScheme;
+        this.approvalLevel = builder.approvalLevel;
+        this.relativePeriodDate = builder.relativePeriodDate;
+        this.userOrgUnit = builder.userOrgUnit;
+        this.apiVersion = builder.apiVersion;
+        this.allowAllPeriods = builder.allowAllPeriods;
+        this.duplicatesOnly = builder.duplicatesOnly;
+    }
+    
+    public static DataQueryRequestBuilder newBuilder()
+    {
+        return new DataQueryRequest.DataQueryRequestBuilder();
+    }
+    
+
+    public static class DataQueryRequestBuilder
+    {
+        private Set<String> dimension;

```

Comment:
For the Builder I don't think you need to duplicate all the fields here, rather just have an instance of DataQueryRequest and set values on that object.